cPP_DSC.adoc

```diff
@@ -2674,243 +2617,254 @@ FCS_CKM.5.1:: The TSF shall derive cryptographic keys with [*selection*: _key ty
-|ISO/IEC 18033-3 (Sub Clause 5.2) [AES] 
+|LEA-CCM
+|LEA in CCM mode with unpredictable, non-repeating nonce, minimum size of 64 bits
+|[selection: 128, 192, 256] bits
+|ISO/IEC 29192-2:2019 (Sub Clause 6.3) [LEA]
 
-ISO/IEC 18033-3:2010 (Sub Clause 5.3) [Camellia]  
+[selection: ISO/IEC 19772:2020 (Clause 7), NIST SP 800-38C] [CCM]
 
-ISO/IEC 18033-3:2010 (Sub Clause 5.4) [SEED] 
+|LEA-GCM
+|LEA in GCM mode with non-repeating IVs using [selection: deterministic, RBG-based] IV construction; the tag must be of length [selection:  96, 104, 112, 120, or 128] bits.
+|[selection: 128, 192, 256] bits
+|ISO/IEC 29192-2:2019 (Sub Clause 6.3) [LEA]
 
-ISO/IEC 29192-2:2019 (Sub Clause 6.3) [LEA] 
+[selection: ISO/IEC 19772:2020 (Clause 10), NIST SP 800-38D] [GCM]
 
-NIST SP 800-38F (Section 6.2) [KW]] 
+|===
 
-|KWP 
-|[*selection*: AES, CAM, SEED, LEA]  in KWP mode 
-|{empty}[*selection*: [.underline]#128 bits, 256 bits#] 
-|ISO/IEC 18033-3 (Sub Clause 5.2) [AES] 
+==== FCS_COP.1/CMAC Cryptographic Operation (CMAC)
 
-ISO/IEC 18033-3:2010 (Sub Clause 5.3) [Camellia]  
+FCS_COP.1/CMAC Cryptographic Operation (CMAC)
 
-ISO/IEC 18033-3:2010 (Sub Clause 5.4) [SEED] 
+FCS_COP.1.1/CMAC:: The TSF shall perform [_CMAC_] in accordance with a specified cryptographic algorithm [_AES-CMAC_] and cryptographic key sizes [*selection*: 128 bits, 192 bits, 256 bits] that meet the following: [*selection*: ISO/IEC 9797-1:2011 sub clause 7.6 (CMAC) and ISO/IEC 18033-3:2010 sub clause 5.2 (AES), NIST SP 800-38B (CMAC) and NIST FIPS 197 (AES)].
 
-ISO/IEC 29192-2:2019 (Sub Clause 6.3) [LEA] 
+==== FCS_COP.1/KeyEncap Cryptographic Operation (Key Encapsulation)
 
-NIST SP 800-38F (Section 6.3) [KWP] 
+FCS_COP.1/KeyEncap Cryptographic Operation (Key Encapsulation)
 
-|AES-CCM
-|AES in CCM mode with unpredictable, non-repeating nonce, minimum size of 64 bits
-|{empty}[*selection*: [.underline]#128 bits, 192 bits, 256 bits#]
-|{empty}[*selection*: [.underline]#ISO/IEC 18033-3 (Sub Clause 5.2), FIPS PUB 197]# [AES]
+FCS_COP.1.1/KeyEncap:: The TSF shall perform _key encapsulation_ in accordance with a specified cryptographic algorithm [*selection*: _cryptographic algorithm_] and cryptographic key sizes [*selection*: _cryptographic key sizes_] that meet the following: [*selection*: _list of standards_].
 
-{empty}[*selection*: [.underline]#ISO/IEC 19772:2020 (Clause 7), NIST SP 800-38C]# [CCM]
+.Key Encapsulation
+[[KeyEncapAlgorithms]]
+[cols=".^1,.^2,.^2",options=header]
+|===
 
-|AES-GCM
-|AES in GCM mode with non-repeating IVs
+|Cryptographic Algorithm 
+|Cryptographic Key Sizes
+|List of Standards
 
-IV length must be equal to 96 bits; the deterministic IV construction method [SP800-38D, Section 8.2.1] must be used; the MAC length t must be one of the values 96, 104, 112, 120, and 128 bits.
-|{empty}[*selection*: [.underline]#128 bits, 192 bits, 256 bits#]
-|{empty}[*selection*: [.underline]#ISO/IEC 18033-3 (Sub Clause 5.2), FIPS PUB 197]# [AES] 
+|KAS1 [RSA-single party] 
```

Comment:
KAS1 as defined in SP 800-56Br2 is a key agreement scheme, not a key encapsulation scheme. It does depend on the RSASVE (Secret Value Encapsulation) primitive which could be seen as an encapsulation. However, the main difference here is that RSASVE generates a random secret as part of the Encapsulation step, rather than taking a key as input. Compare to RSA-OAEP or RSAES-PKCS1-v1_5.

Reply:
@slpotte

Reply:
I've thought about this some.  The reason why it's included as key transport is because the 2nd party contribution is not secret (see the "single-party" qualifier).  I suppose this causes confusion since NIST does list it as a key agreement scheme.  I'm okay with moving it to Key Agreement if you feel it will cause less confusion and has more consistency with NIST is important.  If we leave it here, we should add an app note as to why it is here.  I want to think about for a day or two before moving it.

Reply:
@slpotte I understand your perspective, and I don't want to impose anything on the crypto catalog. I think the main reason NIST classifies it as a KAS rather than KTS, is because the secret value that is transferred from Party U to Party V is not actually the secret key. The secret key is derived from the secret value using a KDF. On the other hand, in RSA OAEP, which NIST defines as a KTS, there is no key derivation step.

Reply:
OK, so it doesn't matter to me which one, FCS_CKM_EXT.7 has KAS2, and I don't know that we have any specific requirement cascade that would cause a problem with it being in either one (unlike when GCM was moved into AEAD), so this is a low risk change from the DSC standpoint, but matching the catalog is preferred.

At this point I'm inclined to leave it as it is (at least in this pull request) and then if the catalog changes we make another pull request to match.

Reply:
Sounds good to me

Reply:
Okay.  No changes to the catalog for now.